golang/lib/networks/network_context.go

```diff
@@ -202,75 +195,44 @@ func (networkCtx *NetworkContext) AddServiceToPartition(
-		}
-		if err := initializingFunc(fp); err != nil {
-			return nil, nil, stacktrace.Propagate(err, "The function to initialize file with ID '%v' returned an error", fileId)
-		}
-		generatedFileAbsFilepathsOnService[fileId] = filepaths.GetAbsoluteFilepathOnServiceContainer()
-	}
-	logrus.Trace("Successfully initialized generated files in suite execution volume")
+func (networkCtx *NetworkContext) StartService(
+		serviceContext *services.ServiceContext,
+		containerConfig *services.ContainerConfig) (map[string]*kurtosis_core_rpc_api_bindings.PortBinding, error) {
 
-	containerRunConfig, err := generateRunConfigFunc(serviceIpAddr, generatedFileAbsFilepathsOnService, staticFileAbsFilepathsOnService)
-	if err != nil {
-		return nil, nil, stacktrace.Propagate(err, "An error occurred getting the container run config")
-	}
+	ctx := context.Background()
 
 	logrus.Tracef("Creating files artifact ID str -> mount dirpaths map...")
 	artifactIdStrToMountDirpath := map[string]string{}
-	for filesArtifactId, mountDirpath := range containerCreationConfig.GetFilesArtifactMountpoints() {
+	for filesArtifactId, mountDirpath := range containerConfig.GetFilesArtifactMountpoints() {
 		artifactIdStrToMountDirpath[string(filesArtifactId)] = mountDirpath
 	}
 	logrus.Tracef("Successfully created files artifact ID str -> mount dirpaths map")
 
 	logrus.Tracef("Starting new service with Kurtosis API...")
 	startServiceArgs := &kurtosis_core_rpc_api_bindings.StartServiceArgs{
-		ServiceId:                  string(serviceId),
-		DockerImage:                containerCreationConfig.GetImage(),
-		UsedPorts:                  containerCreationConfig.GetUsedPortsSet(),
-		EntrypointArgs:             containerRunConfig.GetEntrypointOverrideArgs(),
-		CmdArgs:                    containerRunConfig.GetCmdOverrideArgs(),
-		DockerEnvVars:              containerRunConfig.GetEnvironmentVariableOverrides(),
-		EnclaveDataVolMntDirpath:   containerCreationConfig.GetKurtosisVolumeMountpoint(),
+		ServiceId:                  string(serviceContext.GetServiceID()),
+		DockerImage:                containerConfig.GetImage(),
+		UsedPorts:                  containerConfig.GetUsedPortsSet(),
+		EntrypointArgs:             containerConfig.GetEntrypointOverrideArgs(),
+		CmdArgs:                    containerConfig.GetCmdOverrideArgs(),
+		DockerEnvVars:              containerConfig.GetEnvironmentVariableOverrides(),
+		EnclaveDataVolMntDirpath:   containerConfig.GetKurtosisVolumeMountpoint(),
 		FilesArtifactMountDirpaths: artifactIdStrToMountDirpath,
 	}
 	resp, err := networkCtx.client.StartService(ctx, startServiceArgs)
 	if err != nil {
-		return nil, nil, stacktrace.Propagate(err, "An error occurred starting the service with the Kurtosis API")
+		return map[string]*kurtosis_core_rpc_api_bindings.PortBinding{}, stacktrace.Propagate(err, "An error occurred starting the service with the Kurtosis API")
```

Comment:
I'd personally recommend returning `nil`, rather than emptymap, for errors because with `nil` if the user tries to use the map when `err != nil` they'll get a loud segfault

Reply:
yep! now I remember that you told me that when the method returns an error the remaining returning fields should be always nil, if the type accepts nil value. I don't know why when I was doing this a thought the opposite, maybe because it was something that I usually did in other projects. Thanks